src/python/pants/backend/python/target_types.py

```diff
@@ -1037,13 +1037,27 @@ class SetupPyCommandsField(StringSequenceField):
     )
 
 
+class GenerateSetupField(BoolField):
+    alias = "generate_setup"
```

Comment:
I'd guess most repos are uniform on the value needed, suggesting a subsystem default paired with this to avoid having to set this field in a bunch of BUILD files or write a macro. Maybe worth a follow up before 2.8 gets finalized.

Reply:
Oh, good idea! Will add here.

Reply:
Added.